goalie/mesh_seq.py

```diff
@@ -639,27 +761,26 @@ def check_element_count_convergence(self):
 
     @PETSc.Log.EventDecorator()
     def fixed_point_iteration(
-        self,
-        adaptor: Callable,
-        solver_kwargs: dict = {},
-        adaptor_kwargs: dict = {},
-        **kwargs,
+        self, adaptor, update_params=None, solver_kwargs={}, adaptor_kwargs={}
     ):
         r"""
-        Apply goal-oriented mesh adaptation using a fixed point iteration loop.
+        Apply mesh adaptation using a fixed point iteration loop approach.
 
-        :arg adaptor: function for adapting the mesh sequence. Its arguments are the
-            :class:`~.MeshSeq` instance and the dictionary of solution
-            :class:`firedrake.function.Function`\s. It should return ``True`` if the
+        :arg adaptor: function for adapting the mesh sequence. Its arguments are the mesh
+            sequence and the solution data object. It should return ``True`` if the
             convergence criteria checks are to be skipped for this iteration. Otherwise,
             it should return ``False``.
         :kwarg update_params: function for updating :attr:`~.MeshSeq.params` at each
             iteration. Its arguments are the parameter class and the fixed point
             iteration
-        :kwarg solver_kwargs: a dictionary providing parameters to the solver
-        :kwarg adaptor_kwargs: a dictionary providing parameters to the adaptor
+        :kwarg solver_kwargs: parameters to pass to the solver
+        :type solver_kwargs: :class:`dict`
+        :kwarg adaptor_kwargs: parameters to pass to the adaptor
+        :type adaptor_kwargs: :class:`dict`
+        :returns: solution data object
+        :rtype: :class:`~.ForwardSolutionData`
         """
-        update_params = kwargs.get("update_params")
```

Comment:
Not needed? Just double checking.

Reply:
It's because `update_params` is now explicitly listed as a kwarg, rather than coming from `**kwargs`.